keps/sig-release/20190118-creating-packages.md

```diff
@@ -0,0 +1,417 @@
+     -e KUBE_BUILD_VERSION='1.12.6~beta.0.1+12187918e930d4+dirty' \
+     -e KUBE_BUILD_PLATFORMS='linux/amd64 linux/386' \
+     k8s-pkg-builder
+   ```
+
+- [ ] Create all package configurations for all currently published
+  [packages](#pkg-list) and [platforms](#plat-list)
+
+- [ ] The output of the above tool (a directory tree, holding a set of
+  generated packages) ideally is in a format that can be uploaded to a GCS
+  bucket which can be used as a debian package repo / yum repo.
+
+  If that is not feasible there needs to be a tool that can re-arrange that
+  directory and prepare it for upload. This might also give the opportunity to
+  run certain tools to prepare the directory, e.g. [`dpkg-scanpacakges`,
+  `debpool` or other).
+
+- [ ] Create one tool that does "everything" that needs to be done in the
+  release and can be called by [`anago`][anago].
+
+  Without any special configuration is should do whatever is needed so that it
+  can be called from within [`anago`][anago], however there should be options
+  to specify which components to package, which components to package, get the
+  artefacts by compiling them or pulling them from a staging bucket, ...
+  
+  This might be seen as the
+  "orchestrator tool" and has the following responsibilities:
+  1. determinate the version to be packaged, based on the current git checkout
+     ok `k/k` (a version like in `bazel-genfiles/build/version`)
+  1. do a build (`build/run.sh make $COMPONENT` or such) or download all needed
+     artefacts from from a staging bucket
+  1. run the tool to prepare packages for upload
+  1. upload the packages
+
+
+[anago]: https://github.com/kubernetes/release/blob/master/anago
+
+<!--
+### User Stories [optional]
+
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of the system.
+The goal here is to make this feel real for users without getting bogged down.
+
+#### Story 1
+
+#### Story 2
+
+### Implementation Details/Notes/Constraints [optional]
+
+What are the caveats to the implementation?
+What are some important details that didn't come across above.
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they releate.
+-->
+
+### Risks and Mitigations
+
+- *Risk*: creating another way on how to build packages  
+  *Mitigations*:  
+  - work with the current builder (@sumi) in google
+  - writing super extra great docs so this is the goto way on how to build
+- *Risk*: `nfpm` is not the right tool, it goes away, works differently, ...  
+  *Mitigations*:  
+  - we encapsulate the build of exactly one package, if things change, we can
+    change the internal implementation but keep the external
+    interface
+- *Risk*: We cannot build proper windows packages but community wants/needs them  
+  *Mitigations*:
+  - Talk to people in sig windows, esp. about packaging their components
```

Comment:
packaging for Windows is arguably the easiest; i can help on this front.
[NSIS](https://en.wikipedia.org/wiki/Nullsoft_Scriptable_Install_System) can be used, but to be honest a ZIP file with just a setup script works fine too.